app/plugin.go

```diff
@@ -632,3 +637,63 @@ func (a *App) notifyPluginEnabled(manifest *model.Manifest) error {
 
 	return nil
 }
+
+func (a *App) signaturesFromPathToReader(plugin *pluginSignaturePaths) (signatures []io.ReadSeeker, closeFunc func(), appErr *model.AppError) {
```

Comment:
0/5 for now, but it seems weird to have a `closeFunc` to clean up here. Would it make sense to return the `ReadCloseSeeker` directly and delegate to the caller?

Reply:
@iomodo initially had it like that, however since the API downstream required `io.ReadSeeker` we had to convert `[]ReadCloseSeeker` to `[]ReadSeeker`, as well as iterate over the `Readers` to `Close` them. Seeing all that I suggested we do this instead.
https://github.com/mattermost/mattermost-server/pull/12768/commits/c55f01fcabb10b058b4b7f5807beb5a5a536ba05 for reference.

Reply:
Open to going back but since this is an internal function, I'm personally ok keeping it the way it is. But open to other suggestions.
Thoughts @lieut-data?

Reply:
0/5 on doing anything for this PR, but it does feel a bit clunky either way. I wonder if we could provide the known certificates to this method and just return the zero or one signing key that matches?

Reply:
If I understand question correctly, I don't think we can do that. We can't find which signature corresponds to which public key.

Reply:
There's some discussion over at https://security.stackexchange.com/questions/62916/can-i-get-a-public-key-from-pgp-signature -- would it be relevant in our case?

Reply:
It would. Turns out gpg adds relevant data to the signature so it is possible. But still I'd argue not to do that. `signaturesFromPathToReader` can still return many signatures(in case several public keys have their corresponding signatures). So we will not be able to return zero or one signing key. We will just add complexity to the code with nearly no gain. Thoughts?